packages/expo-gl-cpp/cpp/EXGLNativeMethods.cpp

```diff
@@ -1019,34 +1074,69 @@ _WRAP_METHOD_SIMPLE(disableVertexAttribArray, glDisableVertexAttribArray, index)
 
 _WRAP_METHOD_SIMPLE(enableVertexAttribArray, glEnableVertexAttribArray, index)
 
+template<typename F>
+inline jsi::Value EXGLContext::getActiveInfo(
+        jsi::Runtime& runtime,
+        const jsi::Value* jsArgv,
+        GLenum lengthParam,
+        F &&glFunc) {
+  if (jsArgv[0].isNull()) {
+    return nullptr;
+  }
+
+  EXJS_UNPACK_ARGV(UEXGLObjectId fProgram, GLuint index);
+
+  GLsizei length;
+  GLint size;
+  GLenum type;
+  std::string name;
+  GLint maxNameLength;
+  addBlockingToNextBatch([&] {
+    GLuint program = lookupObject(fProgram);
+    glGetProgramiv(program, lengthParam, &maxNameLength);
+    name.resize(maxNameLength);
+    glFunc(program, index, maxNameLength, &length, &size, &type, &name[0]);
+  });
+
+  if (strlen(name.c_str()) == 0) { // name.length() may be larger
```

Comment:
What's the reason for `name.lenght()` being larger?

Reply:
`glGetProgramiv(program, lengthParam, &maxNameLength);` return max name, actual string might be shorter